lib/service/service.go

```diff
@@ -394,12 +396,12 @@ func (process *TeleportProcess) initAuthService() error {
 			srv.Addr = fmt.Sprintf("%v:%v", process.Config.AdvertiseIP.String(), port)
 		}
 		for !askedToExit {
+			sleepTime := defaults.ServerHeartbeatTTL/2 + utils.RandomDuration(defaults.ServerHeartbeatTTL/10)
+			time.Sleep(sleepTime)
```

Comment:
see comments. it removes a constant "1 second" delay, replacing it with a flexible value from defaults (thus speeding up tests) and skips the first always-failing attempt at connecting a server which hasn't started yet.